### PR TITLE
log warnings but ignore cyclic symlinks

### PR DESCRIPTION
**Pull request by wolfv**
This allows the installation of the slightly broken package `libcufile=1.5.0.59=0 -c nvidia` from the NVidia channel.

### CODE REVIEW

**Comment by wolfv** [comment]
It logs warnings like this:


warning  libmamba Could not check existence: Too many levels of symbolic links (lib/libcufile.so.0)
  Package       Version  Build  Channel               Size
────────────────────────────────────────────────────────────
  Install:
────────────────────────────────────────────────────────────

  + libcufile  1.5.0.59  0      nvidia/linux-64     Cached

  Summary:

  Install: 1 packages

  Total download: 0 B

────────────────────────────────────────────────────────────


Confirm changes: [Y/n] 

Transaction starting
Linking libcufile-1.5.0.59-0
warning  libmamba Could not check existence Too many levels of symbolic links (lib/libcufile.so.0)
Transaction finished

**Comment by wolfv** [comment]
Fixes https://github.com/mamba-org/mamba/issues/2167

File: libmamba/src/core/link.cpp [hunk 1]
```diff
@@ -571,7 +571,8 @@ namespace mamba
             fs::create_directories(dst.parent_path());
         }
 
-        if (fs::exists(dst))
+        std::error_code ec;
+        if (fs::exists(dst, ec) && !ec)
```
Comment (Klaim) [inline libmamba/src/core/link.cpp:575]: Reminder: @Hind-M appropriately pointed in a meeting to prefer [`mamba::lexists(...)`](https://github.com/mamba-org/mamba/blob/main/libmamba/include/mamba/core/util.hpp#L61-L62)
Although we should still use the error-code version until we add an `expected` version.

File: libmamba/src/core/link.cpp [hunk 2]
```diff
@@ -914,7 +918,12 @@ namespace mamba
                 }
                 if (!found)
                 {
-                    if (fs::exists(m_context->target_prefix / files_record[i]))
+                    bool exists = fs::exists(m_context->target_prefix / files_record[i], ec);
+                    if (ec) {
```
Comment (Klaim) [inline libmamba/src/core/link.cpp:922]: This `if` looks like an `else` for the following `if`?

**Pull request merged**